napari/layers/shapes/_shapes_utils.py

```diff
@@ -598,22 +702,27 @@ def triangulate_face(data: npt.NDArray) -> tuple[npt.NDArray, npt.NDArray]:
         Px3 array of the indices of the vertices that will form the
         triangles of the triangulation
     """
-
+    raw_vertices, edges = _normalize_vertices_and_edges(
+        polygon_vertices, close=True
+    )
     if triangulate is not None:
-        len_data = len(data)
-
-        edges = np.empty((len_data, 2), dtype=np.uint32)
-        edges[:, 0] = np.arange(len_data)
-        edges[:, 1] = np.arange(1, len_data + 1)
-        # connect last with first vertex
-        edges[-1, 1] = 0
-
-        res = triangulate({'vertices': data, 'segments': edges}, 'p')
-        vertices, triangles = res['vertices'], res['triangles']
-    elif _is_convex(data):
-        vertices, triangles = _fan_triangulation(data)
+        # if the triangle library is installed, use it because it's faster.
+        res = triangulate(
+            {'vertices': raw_vertices, 'segments': edges}, opts='p'
+        )
+        vertices = res['vertices']
+        raw_triangles = res['triangles']
+        triangles = _cull_triangles_not_in_poly(
+            vertices, raw_triangles, polygon_vertices
+        )
+    elif _is_convex(polygon_vertices):
+        vertices, triangles = _fan_triangulation(polygon_vertices)
```

Comment:
I prefer to call `_is_convex` before `_normalize_vertices_and_edges` as we do not use its output in `_fan_triangulation`

Reply:
Yes that makes sense!